roleplay_economy.py

```diff
@@ -0,0 +1,1098 @@
+            :param item: {'name':'', 'override':{}, 'count': 0}
+            :param key: whether or not to search in key inventory
+            :return:
+            """
+            inv = self.inv_key if key else self.inv
+            for _pos, _item in enumerate(inv):
+                if item['name'] == _item['name'] and item['override'] == _item['override']:
+                    inv[_pos]['count'] += item['count']
+                    if inv[_pos]['count'] <= 0:
+                        del inv[_pos]
+                    return
+            else:
+                if item['count'] > 0:
+                    inv.append(item)
+
+        def override_item(self, item: dict):
+            return {**self.shop[item['name']], **item['override'],
+                    'fields': [(f, v) for f, v in
+                               list({**dict(self.shop[item['name']]['fields']),
+                                     **dict(item['override'].get('fields', []))}.items())
+                               if v]}
+
+        def embed(self):
+            t_embed = Embed(type="rich", colour=16711680)
+            t_embed.title = self.name
+            if self.image:
+                t_embed.set_thumbnail(url=self.image)
+            for (f_name, f_data) in self.fields:
+                t_embed.add_field(name=f_name, value=f_data, inline=False)
+
+            t_str = ""
+
+            for item in self.inv_key[:32]:
+                if item['name'] in self.shop:
+                    _n = item['override'].get('name')
+                    t_str += _n[:23] + '*' if _n else self.shop[item['name']]['name'][:24]
+                else:
+                    t_str += f"{item['name'][:24]:-^24}"
+                t_str = f"{t_str:<24}×{item['count']:5d}\n" if item['count'] > 1 else f"{t_str}\n"
+            if len(self.inv_key) > 32:
+                t_str += f"And {len(self.inv_key)-32} more..."
+
+            if t_str:
+                t_embed.add_field(name="Key Items", value=f"```\n{t_str}```")
+
+            t_str = ""
+
+            for item in self.inv[:32]:
+                if item['name'] in self.shop:
+                    _n = item['override'].get('name')
+                    t_str += _n[:23] + '*' if _n else self.shop[item['name']]['name'][:24]
+                else:
+                    t_str += f"{item['name'][:24]:-^24}"
+                t_str = f"{t_str:<24}×{item['count']:5d}\n" if item['count'] > 1 else f"{t_str}\n"
+            if len(self.inv) > 32:
+                t_str += f"And {len(self.inv)-32} more..."
+
+            if t_str:
+                t_embed.add_field(name="Items", value=f"```\n{t_str}```")
+
+            t_embed.set_footer(text=f"Money: {self.money}")
+
+            return t_embed
+
+    class Shop:
+        user: int
+        page: int
+        category: str
+        time: float = 0
+
+        _mpage: int
+        _gid: str
+        _parent = None
+        _message: Message
+
+        _items: list
+
+        _emoji = "◀🇽▶"
+        _len = 10
```

Comment:
Think these should be instance attributes, same as above

Reply:
although perhaps not _emoji and _len

Reply:
For all intents and purposes, those are type hints.